XLS-0070d-credentials/README.md

```diff
@@ -0,0 +1,573 @@
+
+* Verity sets up her account such that she only wants to interact with properly KYC'd accounts. He trusts Isabel, a credential issuer, to properly vet accounts and issue credentials attesting to that.
+* Alice, who wants to interact with Verity's account, submits her KYC documents to Isabel off-chain. Isabel creates a credential attesting to her trustworthiness on-chain.
+* Alice accepts the credential, making it a valid credential.
+* Alice can now interact with/send funds to Verity.
+
+## 2. On-Ledger Object: `Credential`
+
+This object will be an on-chain representation of a credential, while ensuring that any sensitive details are kept off-chain, to protect the subject's privacy.
+
+### 2.1. Fields
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`LedgerEntryType`| ✔️|`string`|`UInt16`|The ledger object's type (`Credential`).|
+|`Flags`| ✔️|`number`|`UInt32`|Flag values associated with this object.|
+|`Subject`| ✔️|`string`|`AccountID`|The account that the credential is for.|
+|`Issuer`| ✔️|`string`|`AccountID`|The issuer of the credential.|
+|`CredentialType`| ✔️|`string`|`Blob`|A (hex-encoded) value to identify the type of credential from the issuer.|
+|`Expiration`| |`number`|`UInt32`|Optional credential expiration.|
+|`URI`| |`string`|`Blob`|Optional additional data about the credential (such as a link to the VC document).|
+|`OwnerNode`|✔️|`string`|`UInt64`|A hint indicating which page of the sender's owner directory links to this object, in case the directory consists of multiple pages. Note: The object does not contain a direct link to the owner directory containing it, since that value can be derived from the `Account.PreviousTxnID`.|
+|`PreviousTxnID`|✔️|`string`|`Hash256`|The identifying hash of the transaction that most recently modified this object.|
+|`PreviousTxnLgrSeqNumber`|✔️|`number`|`UInt32`|The index of the ledger that contains the transaction that most recently modified this object.|
+
+#### 2.1.1. Object ID
+
+The ID of this object will be a hash that incorporates the `Subject`, `Issuer`, and `CredentialType` fields, combined with a unique space key for `Credential` objects, which will be defined during implementation.
+
+#### 2.1.2. `Flags`
+| Flag Name | Flag Value |
+|-----------|------------|
+|`lsfAccepted`|`0x00010000`|
+
+`lsfAccepted` is on if the subject of the credential has accepted the credential. If it is off, the issuer has the reserve burden; if it is on, the burden moves to the subject.
+
+#### 2.1.3. `CredentialType`
+
+This value is similar to the NFT `Taxon` value, where the value's meaning will be decided by the issuer. It may be the same as a claim in a VC, but could also represent a subset of such a claim.
+
+### 2.2. Account Deletion
+
+The `Credential` object is not a [deletion blocker](https://xrpl.org/docs/concepts/accounts/deleting-accounts/#requirements).
```

Comment:
Does this mean that a `Credential` ledger entry can be automatically deleted alongside either the subject or the issuer accounts?

Reply:
> I think what makes the most sense is that deleting an account also deletes all Credentials where that account is responsible for the reserve, but leaves Credentials where the other party is responsible for the reserve.
> That way, even if an issuer deletes their account, valid credentials that are in use can remain in-ledger but unaccepted credentials are cleaned up.

That's basically what I meant by this section.

> I also don't think it would be totally unreasonable if deleting an account deleted all credentials linked to it regardless of accepted status.

I disagree with this, because credentials already have a mechanism for them to exist without the issuer being on-chain.

Reply:
One of the pre-requisites for account-deletion is that the account has fewer than 1000 ledger objects. 
<img width="952" alt="image" src="https://github.com/user-attachments/assets/58617a14-6ac6-4a98-83eb-7fa3f8050452">


If an Issuer account has 1001 un-accepted (the subject has not sent a `CredentialAccept` transaction) Credentials, can it be deleted?

Reply:
Opinion: From a regulatory point-of-view, non-existent (i.e. hitherto deleted) issuers might pose a problem. I do not have any legal recourse to identify my counter-party in a Payment transaction, because the Issuer has been deleted.

I know that I can choose to revoke (remove `DepositPreauth` objects) for such non-existent credentials issuers. But practically, it is difficult to monitor if Issuer accounts are deleted, in real-time.

Reply:
> If an Issuer account has 1001 un-accepted (the subject has not sent a CredentialAccept transaction) Credentials, can it be deleted?

Evidently not.

> From a regulatory point-of-view, non-existent (i.e. hitherto deleted) issuers might pose a problem. I do not have any legal recourse to identify my counter-party in a Payment transaction, because the Issuer has been deleted.

The account may have been deleted, but the party still exists, and they still hold (and are responsible for) the keypair.